components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/AuthorizationDetailsTypeManagerImpl.java

```diff
@@ -0,0 +1,168 @@
+            throws APIResourceMgtException {
+
+        this.authorizationDetailsTypeMgtDAO.deleteAuthorizationDetailsTypesByApiId(apiId, getTenantId(tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AuthorizationDetailsType getAuthorizationDetailsTypeByApiIdAndTypeId(
+            String apiId, String typeId, String tenantDomain) throws APIResourceMgtException {
+
+        return this.authorizationDetailsTypeMgtDAO
+                .getAuthorizationDetailsTypeByApiIdAndTypeId(apiId, typeId, getTenantId(tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<AuthorizationDetailsType> getAuthorizationDetailsTypes(String filter, String tenantDomain)
+            throws APIResourceMgtException {
+
+        return this.authorizationDetailsTypeMgtDAO
+                .getAuthorizationDetailsTypes(getExpressionNodes(filter, null, null), getTenantId(tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<AuthorizationDetailsType> getAuthorizationDetailsTypesByApiId(String apiId, String tenantDomain)
+            throws APIResourceMgtException {
+
+        return this.authorizationDetailsTypeMgtDAO
+                .getAuthorizationDetailsTypesByApiId(apiId, getTenantId(tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isAuthorizationDetailsTypeExists(String filter, String tenantDomain) throws APIResourceMgtException {
+
+        return CollectionUtils.isNotEmpty(this.getAuthorizationDetailsTypes(filter, tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isAuthorizationDetailsTypeExists(String apiId, String type, String tenantDomain)
+            throws APIResourceMgtException {
+
+        return this.authorizationDetailsTypeMgtDAO
+                .isAuthorizationDetailsTypeExists(apiId, type, getTenantId(tenantDomain));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void replaceAuthorizationDetailsTypes(String apiId, List<String> removedAuthorizationDetailsTypes,
+                                                 List<AuthorizationDetailsType> addedAuthorizationDetailsTypes,
+                                                 String tenantDomain) throws APIResourceMgtException {
+
+        if (CollectionUtils.isNotEmpty(removedAuthorizationDetailsTypes)) {
+            for (String removedType : removedAuthorizationDetailsTypes) {
+                this.authorizationDetailsTypeMgtDAO
+                        .deleteAuthorizationDetailsTypeByApiIdAndType(apiId, removedType, getTenantId(tenantDomain));
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(addedAuthorizationDetailsTypes)) {
+            this.authorizationDetailsTypeMgtDAO
+                    .addAuthorizationDetailsTypes(apiId, addedAuthorizationDetailsTypes, getTenantId(tenantDomain));
+        }
```

Comment:
this should be transactional right? also we can improve this to remove similar AuthorizationDetailsTypes in the both lists